crates/notion-core/src/tool.rs

```diff
@@ -348,3 +355,94 @@ impl Tool for Yarn {
         }
     }
 }
+
+impl Tool for Npm {
+    fn new(session: &mut Session) -> Fallible<Self> {
+        session.add_event_start(ActivityKind::Npm);
+
+        let mut args = args_os();
+        let exe = arg0(&mut args)?;
+        if let Some(ref platform) = session.current_platform()? {
+            session.prepare_image(platform)?;
+
+            if platform.yarn.is_some() {
+                // If Yarn is enabled for the current project, warn the user that they shouldn't be using npm
+                style::display_warning(&String::from("This project is configured to use yarn, npm should be avoided"));
+            }
+            Ok(Self::from_components(&exe, args, &platform.path()?))
+        } else {
+            // Using 'Node' as the tool name since the npm version is derived from the Node version
+            // This way the error message will prompt the user to add 'Node' to their toolchain, instead of 'npm'
+            throw!(NoSuchToolError {
+                tool: "Node".to_string()
+            });
+        }
+    }
+
+    fn from_components(exe: &OsStr, args: ArgsOs, path_var: &OsStr) -> Self {
+        Npm(command_for(exe, args, path_var))
+    }
+
+    fn command(self) -> Command {
+        self.0
+    }
+
+    /// Perform any tasks which must be run after the tool runs but before exiting
```

Comment:
I think you don't need to repeat the rustdoc comment here, since it should already be documented in the trait declaration.